checkov/kubernetes/graph_builder/local_graph.py

```diff
@@ -147,6 +147,9 @@ def _extract_nested_resources_recursive(conf: Dict[str, Any], all_resources: Lis
         if not template or not isinstance(template, dict):
             all_resources.append(conf)
             return
+        if conf.get('kind') == 'Deployment':
+            # means this is a Pod resource nested in a Deployment resource
+            template[PARENT_RESOURCE_KEY_NAME] = conf.get('metadata', {}).get('name', "")
```

Comment:
the idea was to add the resource ID of the parent, then it is easier to query it later.
```python
template[PARENT_RESOURCE_ID_KEY_NAME] = ...
```